### PR TITLE
Fix bug in line charts when the time controls are not ready

### PR DESCRIPTION
**Pull request by ericpgreen2**
Closes https://github.com/rilldata/rill-private-issues/issues/120

The workflow cited in the issue above triggered the `MeasureChart.svelte` component to render with an `undefined` time grain. That crashed the application.

This PR prevents the time series charts from rendering until the `timeControlsStore` has resolved with a defined `interval`.

### CODE REVIEW

File: web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte [hunk 1]
```diff
@@ -204,114 +204,117 @@
...
-          ? EntityStatus.Running
-          : EntityStatus.Idle}
-        on:expand-measure={() => {
-          metricsExplorerStore.setExpandedMeasureName(
-            metricViewName,
-            measure.name,
-          );
-        }}
-      />
+{#if $timeControlsStore.ready}
```
Comment (bcolloran) [inline web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte:207]: Did you investigate whether this conditional can be pushed deeper into the component tree?

I'm fine with going ahead and merging this to clear this bug as a quick stop gap fix, but I suspect we can be more granular here. I'll note that the whole `time-series` folder is one of the ones that [still needs to be cleaned up for strict null checks](https://github.com/rilldata/rill/issues/3424), which is presumably how this bug is possible in the first place.

So I'll happy be happy to dig deeper and clean this up when I get to this file, but in the meantime, I propose adding a FIXME comment before this condition that links back to this PR and the issue and explains briefly that we added this non-granular quick fix, but that we think we can do better. A little reminder like that will help me to be less afraid to refactor this guard out when I get to cleaning up this file

**Review by bcolloran — APPROVED** [review]
approving, but please add a FIXME comment before merging

File: web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte (hunk 1 above)
Comment (ericpgreen2) [reply web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte]: Good call. I just made a more precise conditional, which removed the culpable type error 👍

Comment (bcolloran) [reply web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte]: love it! 👍

**Pull request merged**